### PR TITLE
[analytics] Add subscribe to all

### PR DESCRIPTION
**Pull request by caropinzonsilva**
## ℹ️ What is the context for these changes?
As I was writing more events, I realized that a common use case for analytics would be subscribing to all events (if I want to send all events to my analytics providers for example).

I created a new function `subscribeToAll` that will provide this functionality to the users.

## 🎩 How can this be tophatted?
There are no visual changes, but the tests should be enough to cover the new function.

## ✅ Checklist

- [ ] ~Tested on mobile~ N/A
- [ ] ~Tested on multiple browsers~ N/A
- [ ] ~Tested for accessibility~ N/A
- [x] Includes unit tests
- [ ] ~Updated relevant documentation for the changes (if necessary)~ N/A

### CODE REVIEW

File: packages/shared/src/utils/analytics/ClientAnalytics.test.ts [hunk 1]
```diff
@@ -55,6 +55,22 @@ describe('ClientAnalytics', () => {
     expect(mock2).toHaveBeenCalledWith(eventArgs);
   });
 
+  it('subscriber gets called when using subscribe to all', () => {
+    const eventArgs = {testProps: 'testProps'};
+    const mock1 = vi.fn();
+
+    ClientAnalytics.subscribeToAll(mock1);
```
Comment (caropinzonsilva) [inline packages/shared/src/utils/analytics/ClientAnalytics.test.ts:62]: We only have 1 event for now so I'm only testing that event, but I can add more events here once we start adding more.

**Review by jamiely — APPROVED** [review]
:+1:

File: packages/shared/src/utils/analytics/ClientAnalytics.ts [hunk 2]
```diff
@@ -26,6 +26,18 @@ const subscribe = (
...
+  const eventNamesArray = Object.values(eventNames);
+  eventNamesArray.forEach((eventName) => {
+    subscribe(eventName, (args) =>
+      callbackFunction({
+        eventName,
+        eventArgs: args,
+      }),
+    );
+  });
+};
```
[the remark this answers is not in the document]
Comment (caropinzonsilva) [reply packages/shared/src/utils/analytics/ClientAnalytics.ts:39]: Yeah, it should definitely return it. I'll add it plus the test. Thanks for bringing it up!

[the remark this answers is not in the document]
Comment (caropinzonsilva) [reply packages/shared/src/utils/analytics/ClientAnalytics.ts:49]: Added it here: https://github.com/Shopify/blockchain-components/pull/52/commits/7d336ab5cfade38705cc1ea1d44103c3c3b8c057

[the remark this answers is not in the document]
Comment (QuintonC) [reply packages/shared/src/utils/analytics/ClientAnalytics.ts:49]: Looking at this, when #53 is reviewed we can leverage `.clear()` on the Map of subscribers to clear them all as well.

**Pull request merged**